### DISCUSSION TITLE
Maximum allowed response time for given trial

### DISCUSSION THREAD

**Discussion by sethamarx** [jspsych/jsPsych#2932] [Q&A] [answered]
Hello,

I am writing an experiment that involves visual search circles and want to set a time limit to how many seconds a participant has to give a response before the trial is counted as incorrect. For example, I would like the participant to have 5 seconds from the onset of the visual stimuli to record a response (which is done by either pressing 'f' or 'j') before finishing the trial and moving onto the next.

Does anyone know how this would work?

Thanks

**Comment by jodeleeuw** [comment, marked as answer]
Hi @sethamarx,

You can use the `trial_duration` parameter to control the maximum length of response. 

```js
var trial = {
  type: jsPsychVisualSearchCircle,
  target: 'img/backwardN.gif',
  foil: 'img/normalN.gif',
  fixation_image: 'img/fixation.gif',
  target_present: true,
  set_size: 4,
  trial_duration: 5000
}
```

If no response is given, the RT will be recorded as `null` and correct will be `false`.

**Reply by sethamarx** [reply]
Thank you!
